### PR TITLE
[Android] Use more extensions.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
**Description of the Change**
* Make use of more extension functions provided in Jetpack, such as `String.toUri()`.
* Add an extension function to convert a `long` to a `LocalDateTime`.

**Alternate Designs**
N/A

**Release Notes**
N/A

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectDetailsFragment.kt [hunk 1]
```diff
@@ -364,9 +364,7 @@ class ProjectDetailsFragment : Fragment() {
                 val slideshowBinding = ProjectDetailsSlideshowImageLayoutBinding.inflate(layoutInflater)
                 var bitmap = image.image!!
                 if (scaleImages(bitmap.height, bitmap.width)) {
-                    bitmap = Bitmap.createScaledBitmap(image.image!!,
-                            image.image!!.width * 2,
-                            image.image!!.height * 2, false)
+                    bitmap = bitmap.scale(bitmap.width * 2, bitmap.height * 2, filter = false)
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectDetailsFragment.kt:367]: Why do we do upscale here?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/glide/ScaleBitmapBy2.kt [hunk 2]
```diff
@@ -19,13 +19,14 @@
 package edu.berkeley.boinc.attach.glide
 
 import android.graphics.Bitmap
+import androidx.core.graphics.scale
 import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
 import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
 import java.security.MessageDigest
 
 class ScaleBitmapBy2 : BitmapTransformation() {
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/glide/ScaleBitmapBy2.kt:27]: Why do we need this class at all? Is it possible to remove it and update package with better graphics?

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/glide/ScaleBitmapBy2.kt (hunk 2 above)
Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/glide/ScaleBitmapBy2.kt:27]: It's used in Glide to transform the downloaded image before displaying it.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectDetailsFragment.kt (hunk 1 above)
Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectDetailsFragment.kt:367]: I believe the original developer did it to make the slideshow images more visible.

**Pull request merged**